lib/ClimaCoreTempestRemap/test/mpi_tests/online_remap.jl

```diff
@@ -0,0 +1,149 @@
+using Test
+using BenchmarkTools
+# use these packages for manual inspection of solutions
+# using Plots
+# using ClimaCorePlots
+
+OUTPUT_DIR = mkpath(get(ENV, "CI_OUTPUT_DIR", tempname()))
+
+@testset "distributed online remapping" begin
+    # general setup
+    comms_ctx = ClimaCommsMPI.MPICommsContext()
+    pid, nprocs = ClimaComms.init(comms_ctx)
+
+    comms_ctx_singleton = ClimaComms.SingletonCommsContext()
+
+    # domain
+    radius = 1.0 # unit sphere
+    domain = Domains.SphereDomain(radius)
+
+    # construct source spaces
+    ne_i = 20
+    mesh_i = Meshes.EquiangularCubedSphere(domain, ne_i)
+    topology_i_distr = Topologies.Topology2D(comms_ctx, mesh_i)
+
+    nq_i = 3 # polynomial order for SE discretization
+    quad_i = Spaces.Quadratures.GLL{nq_i}()
+    space_i_distr = Spaces.SpectralElementSpace2D(topology_i_distr, quad_i)
+
+    topology_i_singleton = Topologies.Topology2D(comms_ctx_singleton, mesh_i)
+    space_i_singleton =
+        Spaces.SpectralElementSpace2D(topology_i_singleton, quad_i)
+
+    # construct target spaces
+    ne_o = 5
+    mesh_o = Meshes.EquiangularCubedSphere(domain, ne_o)
+    topology_o_distr = Topologies.Topology2D(comms_ctx, mesh_o)
+
+    nq_o = 4
+    quad_o = Spaces.Quadratures.GLL{nq_o}()
+    space_o_distr = Spaces.SpectralElementSpace2D(topology_o_distr, quad_o)
+
+    topology_o_singleton = Topologies.Topology2D(comms_ctx_singleton, mesh_o)
+    space_o_singleton =
+        Spaces.SpectralElementSpace2D(topology_o_singleton, quad_o)
+
+    # generate test data in the Field format (on non-distributed topo. for non-halo exchange version)
+    field_i_singleton = sind.(Fields.coordinate_field(space_i_singleton).long)
+
+    # global exchange (no buffer fill/send here) - convert to superhalo in next implementation
+    root_pid = 0
+    ClimaComms.gather(comms_ctx, parent(field_i_singleton))
+    field_i_singleton =
+        MPI.bcast(field_i_singleton, root_pid, comms_ctx.mpicomm)
+
+    R = CCTR.generate_map(
+        space_o_singleton,
+        space_i_singleton,
+        target_space_distr = space_o_distr,
+    )
+
+    if ClimaComms.iamroot(comms_ctx)
+        # remap without MPI (for testing comparison) and plot solution
+        field_o_singleton = Fields.zeros(space_o_singleton)
+        # time the serial remapping (can't save result when interpolating)
+        time_serial =
+            @benchmark CCTR.remap!($field_o_singleton, $R, $field_i_singleton)
+        CCTR.remap!(field_o_singleton, R, field_i_singleton)
+    end
+
+    # setup and apply the remap using distributed approach
+    field_o_distr = Fields.zeros(space_o_distr)
+
+    # apply the remapping to field_i_singleton and store the result in field_o_distr
+    # time the distributed remapping (can't save result when interpolating)
+    time_distr =
+        @benchmark (f -> CCTR.remap!($field_o_distr, $R, $field_i_singleton);
+        ClimaComms.barrier($comms_ctx))
```

Comment:
this doesn't look right:

1. `f -> CCTR.remap!($field_o_distr, $R, $field_i_singleton)` is just defining an anonymous function, it isn't actually calling it
2. I don't think it's a good idea to `@benchmark` anything that is calling MPI code, as different processes might run different numbers of times.

Reply:
Is there a better way to quantitatively compare the performance of the MPI vs. serial implementations?